apps/web/src/store/authContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useContext } from 'react';
+import { IOrganizationEntity, IUserEntity, IJwtPayload } from '@novu/shared';
+import { useAuthController } from './useAuthController';
+
+type UserContext = {
+  token: string | null;
+  currentUser: IUserEntity | undefined;
+  currentOrganization: IOrganizationEntity | undefined;
+  organizations: IOrganizationEntity[] | undefined;
+  setToken: (token: string, refetch?: boolean) => void;
+  logout: () => void;
+  jwtPayload?: IJwtPayload;
+};
+
+const AuthContext = React.createContext<UserContext>({
+  token: null,
+  currentUser: undefined,
+  setToken: undefined as any,
+  logout: undefined as any,
+  currentOrganization: undefined as any,
+  organizations: undefined as any,
+  jwtPayload: undefined,
+});
+
+export const useAuthContext = (): UserContext => useContext(AuthContext);
+
+export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
+  const { token, setToken, user, organization, logout, jwtPayload, organizations } = useAuthController();
+
+  return (
+    <AuthContext.Provider
+      value={{
+        currentUser: user,
+        currentOrganization: organization,
+        organizations,
+        token,
+        logout,
+        setToken,
+        jwtPayload,
+      }}
```

Comment:
this object should be memoized

Reply:
this is just moved code and will not be part of this refactoring